snuba/datasets/plans/single_storage.py

```diff
@@ -34,24 +37,31 @@ def __init__(
         self.__query_processors = db_query_processors
         self.__splitters = splitters or []
 
+    @with_span()
     def execute(
         self, query: Query, request_settings: RequestSettings, runner: QueryRunner
     ) -> QueryResult:
         def process_and_run_query(
             query: Query, request_settings: RequestSettings
         ) -> QueryResult:
             for processor in self.__query_processors:
-                processor.process_query(query, request_settings)
+                with sentry_sdk.start_span(
+                    description=type(processor).__name__, op="processor"
+                ):
+                    processor.process_query(query, request_settings)
```

Comment:
I assume this would not work if we tried to add the `@with_span` decorator to the abstract `process_query` method on the QueryProcessor base class in order not to add it wherever we execute a processor since it is abstract.
Unless you already tried I don't think it is worth changing everything to verify this.

Reply:
I made the same assumption based on Python and how it climbs the object hierarchy to do stuff.